perfkitbenchmarker/linux_benchmarks/redis_ycsb_benchmark.py

```diff
@@ -99,12 +99,34 @@ def Run(benchmark_spec):
   groups = benchmark_spec.vm_groups
   redis_vm = groups['workers'][0]
   ycsb_vms = groups['clients']
-  executor = ycsb.YCSBExecutor('redis', **{'redis.host': redis_vm.internal_ip})
-
-  metadata = {'ycsb_client_vms': FLAGS.ycsb_client_vms}
-
-  # This thread count gives reasonably fast load time.
-  samples = list(executor.LoadAndRun(ycsb_vms, load_kwargs={'threads': 4}))
+  # Preparing per client parameters
+  num_ycsb = FLAGS.redis_ycsb_processes
+  num_server = FLAGS.redis_total_num_processes
+  num_client = FLAGS.ycsb_client_vms
+  server_metadata = [{
```

Comment:
These lines and the duplicate/client_vms lines below are pretty complicated - could you please either try to simplify the logic or include more detailed comments?